server/routes/contacts/manage/contact-confirmation/contactConfirmationController.test.ts

```diff
@@ -0,0 +1,183 @@
+    expect($('.govuk-summary-card:eq(0) .govuk-summary-list__key:eq(1)').text().trim()).toStrictEqual('Date of birth')
+    expect($('.govuk-summary-card:eq(0) .govuk-summary-list__key:eq(2)').text().trim()).toStrictEqual('Deceased date')
+
+    // Addresses
+    expect($('h2.govuk-summary-card__title:eq(1)').text().trim()).toStrictEqual('Addresses')
+    expect($('.govuk-summary-card:eq(1) .govuk-summary-card__actions .govuk-link').text().trim()).toContain(
+      'View all addresses',
+    )
+    expect($('.govuk-summary-card:eq(1) .govuk-summary-list__key:eq(0)').text().trim()).toStrictEqual('Address')
+    expect($('.govuk-summary-card:eq(1) .govuk-summary-list__key:eq(1)').text().trim()).toStrictEqual('Type')
+    expect($('.govuk-summary-card:eq(1) .govuk-summary-list__key:eq(2)').text().trim()).toStrictEqual(
+      'Address-specific phone numbers',
+    )
+    expect($('.govuk-summary-card:eq(1) .govuk-summary-list__key:eq(3)').text().trim()).toStrictEqual('Mail')
+    expect($('.govuk-summary-card:eq(1) .govuk-summary-list__key:eq(4)').text().trim()).toStrictEqual('Comments')
+    expect($('.govuk-summary-card:eq(1) .govuk-summary-list__key:eq(5)').text().trim()).toStrictEqual('Dates')
+
+    // Phone numbers
+    expect($('h2.govuk-summary-card__title:eq(2)').text().trim()).toStrictEqual('Phone numbers')
+    expect($('.govuk-summary-card:eq(2) .govuk-summary-list__key:eq(0)').text().trim()).toStrictEqual('Mobile')
+    expect($('.govuk-summary-card:eq(2) .govuk-summary-list__key:eq(1)').text().trim()).toStrictEqual('Business')
+
+    // Email addresses
+    expect($('h2.govuk-summary-card__title:eq(3)').text().trim()).toStrictEqual('Email addresses')
+    expect($('.govuk-summary-card:eq(3) .govuk-summary-list__key:eq(0)').text().trim()).toStrictEqual('Email addresses')
+
+    // Identity numbers
+    expect($('h2.govuk-summary-card__title:eq(4)').text().trim()).toStrictEqual('Identity numbers')
+    expect($('.govuk-summary-card:eq(4) .govuk-summary-list__key:eq(0)').text().trim()).toStrictEqual('Passport number')
+    expect($('.govuk-summary-card:eq(4) .govuk-summary-list__key:eq(1)').text().trim()).toStrictEqual('Driving licence')
+    expect($('.govuk-summary-card:eq(4) .govuk-summary-list__key:eq(2)').text().trim()).toStrictEqual('PNC number')
+
+    // Language details
+    expect($('h2.govuk-summary-card__title:eq(5)').text().trim()).toStrictEqual('Language details')
+    expect($('.govuk-summary-card:eq(5) .govuk-summary-list__key:eq(0)').text().trim()).toStrictEqual('Spoken language')
+    expect($('.govuk-summary-card:eq(5) .govuk-summary-list__key:eq(1)').text().trim()).toStrictEqual(
+      'Needs interpreter',
+    )
+
+    // Radio
+    expect($('.govuk-label:eq(0)').text().trim()).toStrictEqual('Yes, this is the right person')
+    expect($('.govuk-label:eq(1)').text().trim()).toStrictEqual('No, this is not the right person')
+    expect($('.govuk-button').text().trim()).toStrictEqual('Continue')
```

Comment:
I'm not sure testing the headings, radio buttons and continue button labels add much value unless they are optionally appearing?  If you want to keep these assertions, I suggest add `data-qa` instead of using the `eq(1)` style selectors, these are very fragile. If I add a new section then I have to re-order the whole test.

Reply:
Testing on headings, radio buttons and continue button labels removed